packages/discovery/src/discovery/provider/DiscoveryProvider.test.ts

```diff
@@ -0,0 +1,152 @@
+        blockNumber: 0,
+        timestamp: 0,
+      })
+    })
+
+    it('handles simple range', async () => {
+      await discoveryProviderMock.getLogs(address, topics, 5000, 35000)
+      const ranges = rangesFromCalls(providerMock)
+      expect(ranges).toEqual([
+        [5000, 9999],
+        [10000, 19999],
+        [20000, 29999],
+        [30000, 35000],
+      ])
+    })
+
+    it('handles range at boundaries', async () => {
+      await discoveryProviderMock.getLogs(address, topics, 10000, 39999)
+      const ranges = rangesFromCalls(providerMock)
+      expect(ranges).toEqual([
+        [10000, 19999],
+        [20000, 29999],
+        [30000, 39999],
+      ])
+    })
+
+    it('handles range at +/- 1 of boundaries', async () => {
+      await discoveryProviderMock.getLogs(address, topics, 9999, 30000)
+      const ranges = rangesFromCalls(providerMock)
+      expect(ranges).toEqual([
+        [9999, 9999],
+        [10000, 19999],
+        [20000, 29999],
+        [30000, 30000],
+      ])
+    })
+
+    it('handles range where from and to are equal and multiply or range', async () => {
+      await discoveryProviderMock.getLogs(address, topics, 10000, 10000)
+      const ranges = rangesFromCalls(providerMock)
+      expect(ranges).toEqual([[10000, 10000]])
+    })
+
+    it('handles range where from and to are -1 of multiply or range', async () => {
+      await discoveryProviderMock.getLogs(address, topics, 9999, 9999)
+      const ranges = rangesFromCalls(providerMock)
+      expect(ranges).toEqual([[9999, 9999]])
+    })
+
+    it('handles range where from and to are +1 of multiply or range', async () => {
+      await discoveryProviderMock.getLogs(address, topics, 10001, 10001)
+      const ranges = rangesFromCalls(providerMock)
+      expect(ranges).toEqual([[10001, 10001]])
+    })
+
+    it('handles range [0,0]', async () => {
+      await discoveryProviderMock.getLogs(address, topics, 0, 0)
+      const ranges = rangesFromCalls(providerMock)
+      expect(ranges).toEqual([[0, 0]])
+    })
+
+    it('handles range [1,1]', async () => {
+      await discoveryProviderMock.getLogs(address, topics, 1, 1)
+      const ranges = rangesFromCalls(providerMock)
+      expect(ranges).toEqual([[1, 1]])
+    })
+
+    it('handles range inside boundaries', async () => {
+      await discoveryProviderMock.getLogs(address, topics, 1400, 1600)
+      const ranges = rangesFromCalls(providerMock)
+      expect(ranges).toEqual([[1400, 1600]])
+    })
+
+    it('handles getLogsMaxRange undefined (no range)', async () => {
+      providerMock = mockObject<providers.Provider>({
+        getLogs: mockFn().resolvesTo([]),
+      })
+      discoveryProviderMock = new DiscoveryProvider(
+        providerMock,
+        etherscanLikeClientMock,
+        DiscoveryLogger.SILENT,
+        undefined,
+      )
+      discoveryProviderMock.getDeploymentInfo = mockFn().resolvesTo({
+        blockNumber: 0,
+        timestamp: 0,
+      })
```

Comment:
Why is this overwritten once again? It was just assigned to the same thing in `beforeEach`

Reply:
We pass `undefined` for `getLogsMaxRange` to test support for no range. I added a comment in the code.